model/QueueBroker/NewSqlQueueBroker.php

```diff
@@ -0,0 +1,264 @@
+        } catch (Throwable $exception) {
+            $this->logDebug('Schema of ' . $this->getTableName() . ' table already up to date.');
+        }
+        // Create the table
+        $schema = $this->getSchemaProvider()
+            ->setQueueName($this->getQueueName())
+            ->getSchema($schema, $this->getTableName());
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+    }
+
+    /**
+     * Insert a new task into the queue table.
+     */
+    public function push(TaskInterface $task): bool
+    {
+        return (bool)$this->getPersistence()->insert($this->getTableName(), [
+            'id' => $this->getUniquePrimaryKey(),
+            'message' => $this->serializeTask($task),
+            'created_at' => $this->getPersistence()->getPlatForm()->getNowExpression(),
+            'visible' => true,
+        ]);
+    }
+
+    /**
+     * Delete the message after being processed by the worker.
+     *
+     * @param TaskInterface $task
+     */
+    public function delete(TaskInterface $task): void
+    {
+        $this->doDelete($task->getMetadata('NewSqlMessageId'), [
+            'InternalMessageId' => $task->getId(),
+            'NewSqlMessageId' => $task->getMetadata('NewSqlMessageId')
+        ]);
+    }
+
+    public function count(): int
+    {
+        try {
+            return (int)$this->getQueryBuilder()
+                ->select('COUNT(id)')
+                ->from($this->getTableName())
+                ->andWhere('visible = :visible')
+                ->setParameter('visible', true, ParameterType::BOOLEAN)
+                ->execute()
+                ->fetchColumn();
+        } catch (\Exception $e) {
+            $this->logError('Counting tasks failed with MSG: ' . $e->getMessage());
+        }
+
+        return 0;
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    protected function doPop(): void
+    {
+        $this->getPersistence()->getPlatform()->beginTransaction();
+
+        $logContext = $this->getLogContext();
+
+        try {
+            $dbResult = $this->fetchVisibleMessages();
+
+            if ($dbResult) {
+                // set the received messages to invisible for other workers
+                $this->changeMessagesVisibility($dbResult);
+
+                $this->processMessages($dbResult, $logContext);
+            } else {
+                $this->logDebug('No task in the queue.', $logContext);
+            }
+
+            $this->getPersistence()->getPlatform()->commit();
+        } catch (Exception $e) {
+            $this->getPersistence()->getPlatform()->rollBack();
+            $this->logError('Popping tasks failed with MSG: ' . $e->getMessage(), $logContext);
+        }
+    }
+
+    /**
+     * @param string $id
+     * @param array $logContext
+     * @return int
+     */
+    protected function doDelete($id, array $logContext = []): void
```

Comment:
Phpdoc does not match with method signature. If possible we can remove the PHPDoc and use strict types

Reply:
done

Reply:
regarding the `string` for the id, it will break the compatibility with the Abstract

Reply:
Probably the abstract was done in a time we did not have php7. We could fix it there, but OK to keep it like this.